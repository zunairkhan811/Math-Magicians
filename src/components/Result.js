import React from 'react';
import PropTypes from 'prop-types';

export default function Result({ result }) {
  if (!result) return <div className="result"><p>{result}</p></div>;
  return (
    <div className="result">
      <p>
        {result.total}
        {result.operation}
        {result.next}
      </p>
    </div>
  );
}
Result.defaultProps = {
  result: 0,
};
Result.propTypes = {
  result: PropTypes.number,
};
