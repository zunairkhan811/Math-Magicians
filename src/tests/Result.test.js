import React from 'react';
import { render } from '@testing-library/react';
import Result from '../components/Result';

test('renders Result component correctly', () => {
  const result = {
    total: 10,
    operation: '+',
    next: 5,
  };
  const { asFragment } = render(<Result result={result} />);
  expect(asFragment()).toMatchSnapshot();
});
