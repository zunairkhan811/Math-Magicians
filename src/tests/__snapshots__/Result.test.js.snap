// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Result component correctly 1`] = `
<DocumentFragment>
  <div
    class="result"
  >
    <p>
      10+5
    </p>
  </div>
</DocumentFragment>
`;
